src/color.rs

```diff
@@ -157,27 +173,50 @@ impl ExtendedColorType {
             ExtendedColorType::L4 |
             ExtendedColorType::L8 |
             ExtendedColorType::L16 |
+            ExtendedColorType::Bc4 |
             ExtendedColorType::Unknown(_) => 1,
             ExtendedColorType::La1 |
             ExtendedColorType::La2 |
             ExtendedColorType::La4 |
             ExtendedColorType::La8 |
-            ExtendedColorType::La16 => 2,
+            ExtendedColorType::La16 |
+            ExtendedColorType::Bc5 => 2,
             ExtendedColorType::Rgb1 |
             ExtendedColorType::Rgb2 |
             ExtendedColorType::Rgb4 |
             ExtendedColorType::Rgb8 |
             ExtendedColorType::Rgb16 |
-            ExtendedColorType::Bgr8 => 3,
+            ExtendedColorType::Bgr8 |
+            ExtendedColorType::Bc1 |
+            ExtendedColorType::Yuv444 => 3,
+            ExtendedColorType::Yuyv => 3,
             ExtendedColorType::Rgba1 |
             ExtendedColorType::Rgba2 |
             ExtendedColorType::Rgba4 |
             ExtendedColorType::Rgba8 |
             ExtendedColorType::Rgba16 |
-            ExtendedColorType::Bgra8 => 4,
+            ExtendedColorType::Bgra8 |
+            ExtendedColorType::Bc2 |
+            ExtendedColorType::Bc3 => 4,
```

Comment:
I'm not too sure about these values. Yes, there are 4 channels if seen as single values but wasn't this supposed to be used for finding the length of a slice of samples? In a macroblock there is only one sample, the block itself, since it is indivisible without converting it.

Reply:
Yeah, I wasn't sure either. We should probably add a comment explaining what "channels" actually means here.

Reply:
We defined it to agree with `Pixel::CHANNEL_COUNT` and that is used to select a range of same length from a slice of channels. Since we can't subdivide the macro pixel, the most consistent option in my opinion according to that usage would be to define that the 'channel' in that type is the macro pixel itself or at least treat it that way.